crates/turborepo-lockfiles/src/berry/mod.rs

```diff
@@ -55,45 +55,42 @@ pub struct BerryLockfile {
 
 // This is the direct representation of the lockfile as it appears on disk.
 // More internal tracking is required for effectively altering the lockfile
-#[derive(Debug, Clone, Deserialize, Serialize)]
+#[derive(Debug, Clone, Deserialize)]
+#[serde(try_from = "Map<String, Entry>")]
 pub struct LockfileData {
-    #[serde(rename = "__metadata")]
     metadata: Metadata,
-    #[serde(flatten)]
     packages: Map<String, BerryPackage>,
 }
 
-#[derive(Debug, Deserialize, PartialEq, Eq, Hash, PartialOrd, Ord, Serialize, Clone)]
-#[serde(rename_all = "camelCase")]
+#[derive(Debug, PartialEq, Eq, Hash, PartialOrd, Ord, Clone)]
 struct Metadata {
-    version: u64,
+    version: String,
```

Comment:
This will always be an integer, but now that we're avoiding the use of `#[serde(flatten)]` we can parse it as a `String` and it gets to share the same type as `version` in `BerryPackage` which makes our union type easier to work with.